pyanaconda/ui/tui/simpleline/prompt.py

```diff
@@ -0,0 +1,140 @@
+# Class for the Anaconda TUI prompt.
+#
+# Copyright (C) 2016  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+# Author(s):  Vendula Poncova <vponcova@redhat.com>
+#
+
+from pyanaconda.i18n import _, C_
+
+import logging
+log = logging.getLogger("anaconda")
+
+__all__ = ["Prompt"]
+
+
+class Prompt(object):
+    """Class to create a prompt message with options."""
+
+    # Default message of the prompt
+    DEFAULT_MESSAGE = _("Please make a selection from the above")
+
+    # String to use in a prompt when we want users to press the key ENTER.
+    ENTER = _("ENTER")
+
+    # TRANSLATORS: 'q' to quit
+    QUIT = C_('TUI|Spoke Navigation', 'q')
+    QUIT_DESCRIPTION = _("to quit")
+
+    # TRANSLATORS:'c' to continue
+    CONTINUE = C_('TUI|Spoke Navigation', 'c')
+    CONTINUE_DESCRIPTION = _("to continue")
+
+    # TRANSLATORS:'r' to refresh
+    REFRESH = C_('TUI|Spoke Navigation', 'r')
+    REFRESH_DESCRIPTION = _("to refresh")
+
+    # TRANSLATORS:'h' to help
+    HELP = C_('TUI|Spoke Navigation', 'h')
+    HELP_DESCRIPTION = _("to help")
+
+    def __init__(self, message=DEFAULT_MESSAGE):
+        """
+        :param message: the message of the prompt
+        :type message: str|None
+        """
+        self.message = message
+        self.options = dict()
+
+    def set_message(self, message):
+        """Set the prompt message.
+
+        :param message: the message of the prompt
+        :type message: str|None
+        """
+        self.message = message
+
+    def add_option(self, key, description):
+        """Add an option to the prompt.
+        Causes a warning if the option already exists.
+
+        :param key: the key for choosing the option
+        :type key: str
```

Comment:
I'll just note that while a separate `:type key: str` definition is 100% valid, it is also possible to specify the type inline when the type is not ambiguous, like this:

`:param str key: the key for choosing the option`

Or an advanced example from the Sphinx docs:
http://www.sphinx-doc.org/en/1.4.8/domains.html#info-field-lists